File: 2167A/solve.py
import sys
input = sys.stdin.readline

# Link (https://codeforces.com/problemset/problem/2167/A)

def run_tests():
    a,b,c,d = map(int,input().split())

    if a == b == c == d:
        print("YES")
    else:
        print("NO")

def main():
    
    # number of test cases
    t = int(input())
    for _ in range(t):
        run_tests()

if __name__ == "__main__":
    main()

File: 2167A/solve.cpp
#include<iostream>
#include<map>
using namespace std;

// Link (https://codeforces.com/problemset/problem/2167/A)

void run_tests()
{
    int a,b,c,d;
    cin>>a>>b>>c>>d;

    if (a==b && b==c && c==d)
    {
        cout<<"YES\n";
    }
    else
    {
        cout<<"NO\n";
    }
}

int main(){
    int t;
    cin>>t;
    for(int i = 1; i <=t; ++i)
    {
        run_tests();
    }
}